Mathlib/Analysis/Meromorphic/Order.lean

```diff
@@ -730,3 +730,21 @@ theorem codiscrete_setOf_meromorphicOrderAt_eq_zero_or_top (hf : MeromorphicOn f
 alias codiscrete_setOf_order_eq_zero_or_top := codiscrete_setOf_meromorphicOrderAt_eq_zero_or_top
 
 end MeromorphicOn
+
+lemma MeromorphicAt.comp_analyticAt {g : 𝕜 → 𝕜}
+    (hf : MeromorphicAt f (g x)) (hg : AnalyticAt 𝕜 g x) : MeromorphicAt (f ∘ g) x := by
+  obtain ⟨r, hr⟩ := hf
+  by_cases hg' : analyticOrderAt (g · - g x) x = ⊤
+  · -- trivial case: `g` is locally constant near `x`
+    refine .congr (.const (f (g x)) x) ?_
+    filter_upwards [nhdsWithin_le_nhds <| analyticOrderAt_eq_top.mp hg'] with z hz
+    grind
+  · -- interesting case: `g z - g x` looks like `(z - x) ^ n` times a non-vanishing function
+    rw [← Ne, WithTop.ne_top_iff_exists] at hg'
+    obtain ⟨n, hn⟩ := hg'
+    obtain ⟨h, han, hne, heq⟩ := (hg.fun_sub analyticAt_const).analyticOrderAt_eq_natCast.mp hn.symm
+    refine ⟨n * r, (((han.fun_inv hne).pow r).smul (hr.comp hg)).congr ?_⟩
```

Comment:
Is `fun_prop` set up with analysis functions? If so, an even nicer proof could extract the last goal as a have, and then call `fun_prop`.

Reply:
There are some `fun_prop` lemmas for `AnalyticAt`, but it seems to struggle here (probably because it is trying to break up `(g z - g x) ^ r • f (g z)` into its component parts, which is a dead end for this particular proof). OK to leave it as it is for now?